pkg/service/service_test.go

```diff
@@ -763,3 +772,125 @@ func TestServicePlanParameterMarshalling(t *testing.T) {
+			args: args{
+				crd:  "test CRD",
+				name: "testName",
+				devfileObj: parser.DevfileObj{
+					Data: func() data.DevfileData {
+						devfileData, err := data.NewDevfileData(string(data.APIVersion200))
+						if err != nil {
+							t.Error(err)
+						}
+						return devfileData
+					}(),
+					Ctx: devfileCtx.FakeContext(fs, parser.OutputDevfileYamlPath),
+				},
+			},
+			wantErr: false,
+			want: []v1alpha2.Component{{
+				Name: "testName",
+				ComponentUnion: devfile.ComponentUnion{
+					Kubernetes: &devfile.KubernetesComponent{
+						K8sLikeComponent: devfile.K8sLikeComponent{
+							BaseComponent: devfile.BaseComponent{},
+							K8sLikeComponentLocation: devfile.K8sLikeComponentLocation{
+								Inlined: "test CRD",
+							},
+						},
+					},
+				},
+			},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := AddKubernetesComponentToDevfile(tt.args.crd, tt.args.name, tt.args.devfileObj); (err != nil) != tt.wantErr {
+				t.Errorf("AddKubernetesComponentToDevfile() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			got, _ := tt.args.devfileObj.Data.GetComponents(common.DevfileOptions{})
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("GetComponents() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestDeleteKubernetesComponentFromDevfile(t *testing.T) {
+	fs := devfileFileSystem.NewFakeFs()
+
+	type args struct {
+		name       string
+		devfileObj parser.DevfileObj
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+		want    []v1alpha2.Component
+	}{
+		{
+			name: "Case 1: Remove a CRD from devfile.yaml",
+			args: args{
+				name: "testName",
+				devfileObj: parser.DevfileObj{
+					Data: func() data.DevfileData {
+						devfileData, err := data.NewDevfileData(string(data.APIVersion200))
+						if err != nil {
+							t.Error(err)
+						}
+						err = devfileData.AddComponents([]v1alpha2.Component{{
+							Name: "testName",
+							ComponentUnion: devfile.ComponentUnion{
+								Kubernetes: &devfile.KubernetesComponent{
+									K8sLikeComponent: devfile.K8sLikeComponent{
+										BaseComponent: devfile.BaseComponent{},
+										K8sLikeComponentLocation: devfile.K8sLikeComponentLocation{
+											Inlined: "test CRD",
+										},
+									},
+								},
+							},
+						},
+						})
+						if err != nil {
+							t.Error(err)
+						}
+						return devfileData
+					}(),
+					Ctx: devfileCtx.FakeContext(fs, parser.OutputDevfileYamlPath),
+				},
+			},
+			wantErr: false,
+			want:    []v1alpha2.Component{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := DeleteKubernetesComponentFromDevfile(tt.args.name, tt.args.devfileObj); (err != nil) != tt.wantErr {
+				t.Errorf("DeleteKubernetesComponentFromDevfile() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			got, _ := tt.args.devfileObj.Data.GetComponents(common.DevfileOptions{})
```

Comment:
Please handle the error using `t.Errorf()`

Reply:
Done.